### ISSUE TITLE
Please consider support for multi-row Headers for Tables

### ISSUE THREAD

**Issue by wnm3** [remarkjs/remark#1174]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

I also host the OSS site ([MDfromHTML](https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_IBM_MDfromHTML_&d=DwMFaQ&c=jf_iaSHvJObTbx-siA1ZOg&r=EYX3AEyQAwmdbWQyzpxSNg&m=YQ7MINbfYJYaXPQ5SszXp2RLpM8RpTf5pTlBA0EHbjCuPhYCGacjZtm8b-G1ogT8&s=WfiYvgZvIM5-L1QjFmvSa9KDbtf26eaDJ6qVkhWaDkY&e=)) to convert HTML to Markdown. I’ve recently been working to improve table rendering from a bunch of HTML tables I’d put into a test page [here](https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_IBM_MDfromHTML_blob_master_Remark_src_test_resources_conversions_html_tablemulti.html&d=DwMFaQ&c=jf_iaSHvJObTbx-siA1ZOg&r=EYX3AEyQAwmdbWQyzpxSNg&m=YQ7MINbfYJYaXPQ5SszXp2RLpM8RpTf5pTlBA0EHbjCuPhYCGacjZtm8b-G1ogT8&s=Ar7xsiFakDUybGwN_1OtP9JXTq27abWoe3dmGV6pYhc&e=). Some of these provide multiple header rows. I’ve supported this and Fletcher Penney’s MultiMarkdown Viewer supports this format as well. Note: for colspans and rowspans in HTML I elected to replicate the value so the table in HTML with multiple headers originally looked like this:
![image](https://github.com/remarkjs/remark/assets/1359024/8c555d6c-1c59-413d-bebd-4f2b2d0f39d1)
I did this replication as one goal I have is to associate row and column headers with each cell…

Below is a screen shot of the markdown and rendering I’d hoped would occur:
![image](https://github.com/remarkjs/remark/assets/1359024/afe827b2-8e04-403e-a86c-02860913e853)

Unfortunately, the plugin renders the multi-line headers by ignoring all but the last line (directly above the header separator line with pipes and hyphens):

![image](https://github.com/remarkjs/remark/assets/1359024/facfbcaa-233e-438b-8ba9-3e0f4263a22b)

Do you suppose you could add support for multi-row headers? Thank you in advance for your consideration. I will want to request the same for github’s markdown processing so if you know who I should contact please let me know. Thanks.

### Solution

When encountering a table, read ahead to reach the header separator row (pipes, hyphens, spaces, colons...) then apply the alignment gleaned there to all rows above the separator to create the table header rows, then process the rows following the separator as table body rows.

### Alternatives

It could be possible to consolidate all cells for multi-row headers by column using `<br>` or something, but this loses the boundary for explicit headers.

**Comment by ChristianMurphy** [comment]
Welcome @wnm3! 👋 
Thanks for sharing your idea!

### TL;DR
Our table implementation strictly follows GitHub Flavored Markdown (gfm), diverging from the standard, even as an option is a non-goal for the project.
This could be filed with the gfm team at https://github.com/github/cmark-gfm or https://github.com/orgs/community/discussions, and once approved and merged into the GFM standard, we will match the standard.

### Context:
Remark is a wrapper around lower level utilities like https://github.com/micromark/micromark (CommonMark implementation), https://github.com/micromark/micromark-extension-gfm (GFM implementation), and https://github.com/micromark/micromark-extension-mdx (MDX implementation).
Designed to make it easier to work with the abstract syntax tree (more on that here: https://unifiedjs.com/learn/).

CommonMark (https://spec.commonmark.org/), GitHub flavored Markdown (https://github.github.com/gfm/), and MDX (https://github.com/mdx-js/specification) each follow a spec/standard.
These existing plugins will not add features from outside their respective standards.

That said, you and others are welcome to looking into creating your own extension if wanted (guide: https://github.com/micromark/micromark#extending-markdown).
Though know that creating yet another markdown flavor is both a complex endeavor technically, and would require training authors on the new standard.

Another possibility is reaching out to the standards groups themselves with the idea, and adding a feature to the standard. CommonMark has a standards discussion forum (https://talk.commonmark.org/), GitHub responses both to issues and discussions (https://github.com/github/cmark-gfm and https://github.com/orgs/community/discussions), and MDX has a discussion forum (https://github.com/orgs/mdx-js/discussions)

Your particular request is around tables, which is a part of the GFM standard.
